pkg/reconciler/extensions.go

```diff
@@ -0,0 +1,75 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extension"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+type extensions struct {
+	extensions []extension.Extension
+}
+
+func (es *extensions) register(e extension.Extension) {
+	es.extensions = append(es.extensions, e)
+}
+
+func (es *extensions) len() int {
+	return len(es.extensions)
+}
+
+func (es *extensions) get(idx int) extension.Extension {
```

Comment:
Why do these exist? They are unexported, so every caller that can access these methods can also access the `extensions` field directly

Reply:
My idea here was to encapsulate the extensions registration/execution.